cluster.go

```diff
@@ -226,31 +253,54 @@ func (c *clusterClient) _refresh() (err error) {
 		if _, ok := conns[addr]; ok {
 			conns[addr] = cc
```

Comment:
The role of a connection can changed overtime, therefore we should not override the whole field from the old map, we should keep the new role instead.

Reply:
Oh, i missed that point. [d115a78](https://github.com/redis/rueidis/pull/402/commits/d115a789988ccb28a17add44fb5422930d2a4442) Thank you!